hawkular-alerts-bus/src/main/java/org/hawkular/alerts/bus/init/CacheManager.java

```diff
@@ -140,4 +167,49 @@ private synchronized void updateActiveIds() {
         }
     }
 
+    private boolean isNewAvailId(String availId) {
+        if (activeAvailabityIds == null) {
+            return true;
+        }
+        return !activeAvailabityIds.contains(availId);
+    }
+
+    private boolean isNewDataId(String dataId) {
+        if (activeDataIds == null) {
+            return true;
+        }
+        return !activeDataIds.contains(dataId);
+    }
+
+    private void publishCommand(String tenantId, String dataId) throws Exception {
+        String type;
+        String metricId;
+        if (dataId.startsWith(AVAILABILITY)) {
+            type = AVAILABILITY;
+            metricId = dataId.substring(AVAILABILITY.length() + 1);
+        } else if (dataId.startsWith(GAUGE)) {
+            type = GAUGE;
+            metricId = dataId.substring(GAUGE.length() + 1);
+        } else if (dataId.startsWith(COUNTER)) {
+            type = COUNTER;
+            metricId = dataId.substring(COUNTER.length() + 1);
+        } else if (dataId.startsWith(COUNTER_RATE)) {
+            type = COUNTER_RATE;
+            metricId = dataId.substring(COUNTER_RATE.length() + 1);
+        } else if (dataId.startsWith(STRING)) {
+            type = STRING;
+            metricId = dataId.substring(STRING.length() + 1);
+        } else if (dataId.startsWith(GAUGE_RATE)) {
+            type = GAUGE_RATE;
+            metricId = dataId.substring(GAUGE_RATE.length() + 1);
+        } else {
+            if (log.isDebugEnabled()) {
+                log.debug("DataId " + dataId + " doesn't have a valid metrics type. Skipping publishing command.");
+            }
+            return;
+        }
+        PublishCommandMessage msg = new PublishCommandMessage(PUBLISH_COMMAND, tenantId,
+                asList(new PublishCommandMessage.MetricKey(type, metricId)));
+        publishCommandSender.send(msg);
+    }
```

Comment:
Yes, I know what you mean.
I decided to make them explicit because I think in easy to understand the logic, even if some prefix like "availability" can be deducted from the wrapped message type.
I think make them explicit helps to maintain them in the future without paying a lot.